### PR TITLE
En 3794/wasm vm integration

### PR DESCRIPTION
**Pull request by sasurobert**
WASM VM integration. WASM VM binary should be in $USR_HOME/elrond-vm-binaries/

### CODE REVIEW

File: core/libLocator/vmLocator_windows.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+)
+
+const libName = "libhera.dll"
+
+func WASMLibLocation() string {
+	usr, err := user.Current()
+	if err != nil {
+		return ""
+	}
+	return usr.HomeDir + "/elrond-vm-binaries/" + libName
```
Comment (camilbancioiu) [inline core/libLocator/vmLocator_windows.go]: Should replace slashes with backslashes. Currently, the resulting path looks like this on Windows: `C:\Users\User/elrond-vm-binaries/libhera.dll`

Not critical, but not recommended.

File: integrationTests/vm/wasm/wasmVM_test.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+func Benchmark_VmDeployWithCPUCalculateAndExecute(b *testing.B) {
+	runWASMVMBenchmark(b, "./cpucalculate_ewasmified.wasm", b.N, 8000)
+}
+
+func Benchmark_VmDeployWithStringConcatAndExecute(b *testing.B) {
+	runWASMVMBenchmark(b, "./stringconcat_ewasmified.wasm", b.N, 10000)
+}
+
+func runWASMVMBenchmark(tb testing.TB, fileSC string, numRun int, testingValue uint64) {
```
Comment (camilbancioiu) [inline integrationTests/vm/wasm/wasmVM_test.go]: This function runs a benchmark on WAVM specifically. Its name should rather be `runWAVMBenchmark()` to reflect this.

File: integrationTests/vm/wasm/wasmVM_test.go [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+	for i := 0; i < numRun; i++ {
+		tx.Nonce = aliceNonce
+
+		_ = txProc.ProcessTransaction(tx, round)
+
+		aliceNonce++
+	}
+}
+
+func TestVmDeployWithTransferAndExecuteERC20(t *testing.T) {
```
Comment (camilbancioiu) [inline integrationTests/vm/wasm/wasmVM_test.go]: Might want to add `_WABT` at the end of the function's name, to differentiate it from identical tests performed on the other VMs.

File: integrationTests/vm/testInitializer.go [hunk 4]
```diff
@@ -306,12 +308,18 @@ func GetAccountsBalance(addrBytes []byte, accnts state.AccountsAdapter) *big.Int
...
 
 func GetIntValueFromSC(accnts state.AccountsAdapter, scAddressBytes []byte, funcName string, args ...[]byte) *big.Int {
+	returnedVals := GetBytesValueFromSC(accnts, scAddressBytes, funcName, args...)
+
+	return big.NewInt(0).SetBytes(returnedVals)
+}
+
+func GetBytesValueFromSC(accnts state.AccountsAdapter, scAddressBytes []byte, funcName string, args ...[]byte) []byte {
 	vmContainer, _ := CreateVMsContainerAndBlockchainHook(accnts)
 	scgd, _ := smartContract.NewSCDataGetter(vmContainer)
```
Comment (miiu96) [inline integrationTests/vm/testInitializer.go]: you should rename this with scdg

File: integrationTests/vm/wasm/wasmVM_test.go [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/integrationTests/vm"
+	"github.com/ElrondNetwork/elrond-go/process/factory"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestVmDeployWithTransferAndGasShouldDeploySCCode(t *testing.T) {
```
Comment (miiu96) [inline integrationTests/vm/wasm/wasmVM_test.go]: test fail on my machine with latest stable version from hera repository (https://github.com/ElrondNetwork/hera/releases)
image

Comment (camilbancioiu) [reply integrationTests/vm/wasm/wasmVM_test.go]: Likely due to newly added Gas Metering on WABT.  I'll look into it.

Comment (sasurobert) [reply integrationTests/vm/wasm/wasmVM_test.go]: Yes, but currently we do not use the latest from hera, but an earlier release.

File: integrationTests/vm/wasm/wasmVM_test.go (hunk 3 above)
Comment (sasurobert) [reply integrationTests/vm/wasm/wasmVM_test.go]: done

File: integrationTests/vm/wasm/wasmVM_test.go (hunk 2 above)
Comment (sasurobert) [reply integrationTests/vm/wasm/wasmVM_test.go]: done

File: core/libLocator/vmLocator_windows.go (hunk 1 above)
Comment (sasurobert) [reply core/libLocator/vmLocator_windows.go]: done.

File: integrationTests/vm/testInitializer.go (hunk 4 above)
Comment (sasurobert) [reply integrationTests/vm/testInitializer.go]: done.

**Pull request merged**